sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2550,6 +2550,13 @@ object functions {
    * @group datetime_funcs
    * @since 1.5.0
    */
+  def dayofweek(e: Column): Column = withExpr { DayOfWeek(e.expr) }
+
+  /**
+   * Extracts the day of the month as an integer from a given date/timestamp/string.
+   * @group datetime_funcs
+   * @since 2.3.0
```

Comment:
Revert back to `1.5.0` here and modify `@since` to `2.3.0` in the doc for `dayofweek`.